src/Norway.js

```diff
@@ -0,0 +1,292 @@
+    const point = this.getBoundingBoxCenter(box);
+
+    const parentRect = ReactDOM.findDOMNode(this).getBoundingClientRect();
+
+    var scrollTop = document.documentElement.scrollTop || document.body.scrollTop;
+    var scrollLeft = document.documentElement.scrollLeft || document.body.scrollLeft;
+
+    return {
+      x: parentRect.x + point[0] + scrollLeft,
+      y: parentRect.y + point[1] + scrollTop
+    };
+  }
+
+  getCountyCenter(countyIndex) {
+    const selection = this.getCounty(countyIndex);
+    return this.getCenter(selection);
+  }
+
+  fadeInTooltip() {
+    d3.select(".tooltip")
+      .transition()
+      .duration(200)
+      .style("opacity", 0.9);
+  }
+
+  positionTooltip(text, x, y) {
+    d3.select(".tooltip")
+      .html(text)
+      .style("left", x + "px")
+      .style("top", y + "px");
+  }
+
+  showCountyTooltip(countyIndex, event) {
+    this.fadeInTooltip();
+
+    const { x, y } = this.getCountyCenter(countyIndex);
+    const text = this.getCountyName(countyIndex);
+
+    this.positionTooltip(text, x, y);
+
+    // Propagate event
+    const { onMouseOver } = this.props;
+    onMouseOver(event, this.getCountyName(countyIndex));
+  }
+
+  hideTooltip(event) {
+    d3.select(".tooltip")
+      .transition()
+      .duration(500)
+      .style("opacity", 0);
+    // Propagate event
+    const { onMouseOut } = this.props;
+    onMouseOut(event);
+  }
+
+  hideCountyTooltip(countyIndex, event) {
+    this.hideTooltip(event);
+    console.log("Hide Tooltip : " + this.getCountyName(countyIndex));
+  }
+
+  /**
+   *  Municipality
+   */
+  getMunicipalityCenter(municipalityIndex) {
+    const selection = this.getMunicipality(municipalityIndex);
+    return this.getCenter(selection);
+  }
+
+  getMunicipality(municipalityIndex) {
+    const key = "#" + this.makeId("municipality", municipalityIndex);
+    return d3.select(key);
+  }
+
+  getMunicipalityNumber(municipalityIndex) {
+    return this.state.municipalities[municipalityIndex].properties.ID_1;
+  }
+
+  getMunicipalityName(municipalityIndex) {
+    return this.state.municipalities[municipalityIndex].properties.NAME_2;
+  }
+
+  handleMunicipalityClick(municipalityIndex, event) {
+    console.log("Clicked : " + this.getMunicipalityName(municipalityIndex));
+  }
+
+  showMunicipalityTooltip(municipalityIndex, event) {
+    this.fadeInTooltip();
+
+    const { x, y } = this.getMunicipalityCenter(municipalityIndex);
+    const text = this.getMunicipalityName(municipalityIndex);
+
+    this.positionTooltip(text, x, y);
+
+    // Propagate event
+    const { onMouseOver } = this.props;
+    onMouseOver(event, this.getMunicipalityName(municipalityIndex));
+
+    console.log("Show Tooltip : " + this.getMunicipalityName(municipalityIndex));
+  }
+
+  hideMunicipalityTooltip(municipalityIndex, event) {
+    this.hideTooltip(event);
+    console.log("Hide Tooltip : " + this.getMunicipalityName(municipalityIndex));
+  }
+
+  render() {
```

Comment:
You'll need a shouldComponentUpdate for this, or alternatively a useEffect hook if you were to make this a function-component. As you only want render() to be re-called when `id, width, height, this.state.counties, or this.state.municipalities` change. Once again, making the svg use percent-based units could help with not needing explicit height/width values.